webactions.md

```diff
@@ -0,0 +1,546 @@
+      "content-type": "application/json",
+      "host": "172.17.0.1",
+      "user-agent": "curl/7.43.0"
+    },
+    "__ow_path": ""
+  }
+}
+```
+
+For convenience, query parameters, form data, and JSON object body entities are all treated as dictionaries, and their values are directly accessible as action input properties. This is not the case for web actions that opt to handle HTTP request entities more directly or when the web action receives an entity that is not a JSON object.
+
+Here is an example of using a "text" content-type with the same example shown above.
+
+```bash
+$ curl https://${APIHOST}/api/v1/web/my-namespace/demo/hello.json -H 'Content-Type: text/plain' -d "Jane"
+{
+  "response": {
+    "__ow_method": "post",
+    "__ow_headers": {
+      "accept": "*/*",
+      "connection": "close",
+      "content-length": "4",
+      "content-type": "text/plain",
+      "host": "172.17.0.1",
+      "user-agent": "curl/7.43.0"
+    },
+    "__ow_path": "",
+    "__ow_body": "Jane"
+  }
+}
+```
+
+## Content extensions
+
+A content extension is generally required when invoking a web action; the absence of an extension assumes `.http` as the default. The fully qualified name of the action must include its package name, which is `default` if the action is not in a named package.
+
+
+## Protected parameters
+
+Action parameters are protected and treated as immutable. Parameters are automatically finalized when enabling web actions.
+
+```bash
+$ nim action create /my-namespace/demo/hello hello.js \
+      --parameter name Jane \
+      --web true
+```
+
+The result of these changes is that the `name` is bound to `Jane` and cannot be overridden by query or body parameters because of the final annotation. This secures the action against query or body parameters that try to change this value whether by accident or intentionally.
+
+## Securing web actions
+
+**[[NH: I don't know what to do with this section, please advise.]]**
+By default, a web action can be invoked by anyone having the web action's invocation URL. Use the `require-whisk-auth` [web action annotation](annotations.md#annotations-specific-to-web-actions) to secure the web action. When the `require-whisk-auth` annotation is set to `true`, the action will authenticate the invocation request's Basic Authorization credentials to confirm they represent a valid OpenWhisk identity.  When set to a number or a case-sensitive string, the action's invocation request must include a `X-Require-Whisk-Auth` header having this same value. Secured web actions will return a `Not Authorized` when credential validation fails.
+
+Alternatively, use the `--web-secure` flag to automatically set the `require-whisk-auth` annotation.  When set to `true` a random number is generated as the `require-whisk-auth` annotation value. When set to `false` the `require-whisk-auth` annotation is removed.  When set to any other value, that value is used as the `require-whisk-auth` annotation value.
```

Comment:
The `nim` command supports `--web-secure` but there is a difference.  A value of `true` just sets the annotation to `true`.   It does not generate a random number.

Reply:
Rewrote this section and added subsections.